cate/ds/esa_cci_odp_os.py

```diff
@@ -0,0 +1,1376 @@
+
+    def local_dataset_dir(self):
+        return os.path.join(get_data_store_path(), self._datasource_id)
+
+    def local_metadata_dataset_dir(self):
+        return os.path.join(get_metadata_store_path(), self._datasource_id)
+
+    def _find_files(self, time_range):
+        requested_start_date, requested_end_date = time_range if time_range else (None, None)
+        asyncio.run(self._init_file_list())
+        if requested_start_date or requested_end_date:
+            selected_file_list = []
+            for file_rec in self._file_list:
+                start_time = file_rec[1]
+                ok = False
+                if start_time:
+                    if requested_start_date and requested_end_date:
+                        ok = requested_start_date <= start_time <= requested_end_date
+                    elif requested_start_date:
+                        ok = requested_start_date <= start_time
+                    elif requested_end_date:
+                        ok = start_time <= requested_end_date
+                if ok:
+                    selected_file_list.append(file_rec)
+        else:
+            selected_file_list = self._file_list
+        return selected_file_list
+
+    def open_dataset(self,
+                     time_range: TimeRangeLike.TYPE = None,
+                     region: PolygonLike.TYPE = None,
+                     var_names: VarNamesLike.TYPE = None,
+                     protocol: str = None,
+                     monitor: Monitor = Monitor.NONE) -> Any:
+        time_range = TimeRangeLike.convert(time_range) if time_range else None
+        var_names = VarNamesLike.convert(var_names) if var_names else None
+
+        selected_file_list = self._find_files(time_range)
+        if not selected_file_list:
+            raise self._empty_error(time_range)
+
+        files = self._get_urls_list(selected_file_list, _ODP_PROTOCOL_OPENDAP)
+        try:
+            return open_xarray_dataset(files, region=region, var_names=var_names, monitor=monitor)
+        except HTTPError as e:
+            raise self._cannot_access_error(time_range, region, var_names,
+                                            cause=e) from e
+        except (URLError, socket.timeout) as e:
+            raise self._cannot_access_error(time_range, region, var_names,
+                                            cause=e, error_cls=NetworkError) from e
+        except OSError as e:
+            raise self._cannot_access_error(time_range, region, var_names,
+                                            cause=e) from e
+
+    @staticmethod
+    def _get_urls_list(files_description_list, protocol) -> Sequence[str]:
+        """
+        Returns urls list extracted from reference esgf specific files description json list
+        :param files_description_list:
+        :param protocol:
+        :return:
+        """
+        return [file_rec[4][protocol].replace('.html', '') for file_rec in files_description_list]
+
+    def _make_local(self,
```

Comment:
This method is way too large. Split into smaller peaces, so this code becomes human-readable and testable.

Reply:
Why is this here at all, when there is `make_local(self, ...)` too?